src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+        }
+    }
+}
+
+
+
+impl TokenClassificationOption {
+    /// Instantiate a new token sequence classification model of the supplied type.
+    ///
+    /// # Arguments
+    ///
+    /// * `model_type` - `ModelType` indicating the model type to load (must match with the actual data to be loaded)
+    /// * `p` - `tch::nn::Path` path to the model file to load (e.g. model.ot)
+    /// * `config` - A configuration (the model type of the configuration must be compatible with the value for
+    /// `model_type`)
+    ///
+    pub fn new(model_type: ModelType, p: &tch::nn::Path, config: &ConfigOption) -> Self {
+        match model_type {
+            ModelType::Bert => {
+                if let ConfigOption::Bert(config) = config {
+                    TokenClassificationOption::Bert(BertForTokenClassification::new(p, config))
+                } else {
+                    panic!("You can only supply a BertConfig for Bert!");
+                }
+            },
+            ModelType::DistilBert => {
+                if let ConfigOption::DistilBert(config) = config {
+                    TokenClassificationOption::DistilBert(DistilBertForTokenClassification::new(p, config))
+                } else {
+                    panic!("You can only supply a DistilBertConfig for DistilBert!");
+                }
+            }
+            ModelType::Roberta => {
+                if let ConfigOption::Bert(config) = config {
+                    TokenClassificationOption::Roberta(RobertaForTokenClassification::new(p, config))
+                } else {
+                    panic!("You can only supply a BertConfig for Roberta!");
+                }
+            }
+        }
+    }
+
+    /// Returns the `ModelType` for this TokenClassificationOption
+    pub fn model_type(&self) -> ModelType {
+        match *self {
+            Self::Bert(_) => ModelType::Bert,
+            Self::Roberta(_) => ModelType::Roberta,
+            Self::DistilBert(_) => ModelType::DistilBert
+        }
+    }
+
+    /// Interface method to forward_t() of the particular models.
+    pub fn forward_t(&self,
+                     input_ids: Option<Tensor>,
+                     mask: Option<Tensor>,
+                     token_type_ids: Option<Tensor>,
+                     position_ids: Option<Tensor>,
+                     input_embeds: Option<Tensor>,
+                     train: bool) -> (Tensor, Option<Vec<Tensor>>, Option<Vec<Tensor>>)  {
+        match *self {
+            Self::Bert(ref model) => model.forward_t(input_ids,mask,token_type_ids,position_ids,input_embeds,train),
+            Self::DistilBert(ref model) => model.forward_t(input_ids,mask,input_embeds,train).expect("Error in distilbert forward_t"),
+            Self::Roberta(ref model) => model.forward_t(input_ids,mask,token_type_ids,position_ids,input_embeds,train),
+        }
+
+    }
+
+}
+
+/// # Holds the label mapping for the model, supports different model types.
+pub enum LabelMapping {
```

Comment:
Maybe we could instead harmonize the `config.id2label` throughout the models to all be `HashMap<i64,String>`. This should not break any code (was a mistake on my side) and would allow skipping this enum and the methods associated with it.

Reply:
Ah, yes, if there's no need for a different key size then that would indeed be much simpler. I assumed you opted for i32 for DistilBert to save memory or something :) (which probably indeed doesn't make that much sense now that I come to think of it as the label map probably won't get very big anyway).

Reply:
DistilBert was the first model I implemented and I wanted to be memory conscious at that level. I then noticed that most tensor operations work with f64/i64 and ended-up casting a lot from i8/16 to i64 so I defaulted to the larger values in later models. The mapping for labels is so small usually (less than 100 labels) that this should not matter anyway.